File: backend/dalathasfarm/src/main/java/com/example/dalathasfarm/models/Category.java
package com.example.dalathasfarm.models;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "categories")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, unique = true)
    private String name;

    private String description;

    @Column(unique = true)
    private String thumbnail;
}

File: android/lib/models/category.dart
class Category {
    final int id;
    final String name;
    final String description;
    final String thumbnail;

    Category({
      required this.id,
      required this.name,
      required this.description,
      required this.thumbnail,
    });

    factory Category.fromJson(Map<String, dynamic> json) {
      return Category(
        id: json['id'],
        name: json['name'],
        description: json['description'],
        thumbnail: json['thumbnail'],
      );
    }
}